libtiledbsc/include/tiledbsc/soma.h

```diff
@@ -0,0 +1,46 @@
+#ifndef SOMA_H
+#define SOMA_H
+
+#include <tiledb/tiledb>
+#include <tiledb/tiledb_experimental>
+
+namespace tiledbsc {
+using namespace tiledb;
+
+class SOMA {
+   public:
+    //===================================================================
+    //= public static
+    //===================================================================
+    static SOMA open(std::string_view uri);
+
+    //===================================================================
+    //= public non-static
+    //===================================================================
+    SOMA(std::string_view uri);
+
+    std::map<std::string, std::string> list_arrays();
```

Comment:
I believe this is a map from group-element name to group-element URI -- let's say that as a comment

Reply:
Further should this be an `unordered_map`?

Reply:
I'll add more comments and update to `unordered_map`.

Reply:
why not ordered map?

Reply:
Unordered maps are more efficient because they don't sort the keys. It probably doesn't matter much in this case, since the maps should be small.

I made it ordered to begin with because I liked seeing sorted names when printing the map, but sorting can be done later.